UCSD/campus-mobile
# Bug/827 authenticated cards

## Summary

Resolves #827.

## Changelog

[General] [Fix] - Fixing/allowing logged in users to save their card order and card states preferences to their profile.


## Test Plan

There are 3 different 'accounts' to test: visitor, student, and staff:

1. Testing Visitor
    1. Go to 'Cards' page in the 'Profile' page
        - Remember the card order and states
        - Change card(s) order and states
        - Go back to the home page and see if those changes are reflected
    2. On the home page, hide some card(s)
        - Go back to the 'Cards' page and see if those changes are reflected
    3. To see if the changes stay, remove the app from memory and open it back up
    4. Change the order and states of the cards back to what they were at the start
2. Testing Student
    1. Upon logging in, go to 'Cards' page in the 'Profile' page and make sure that all the cards are turned on and the order is as follows:
        - ![Screen Shot 2022-02-15 at 2 14 40 PM](https://user-images.githubusercontent.com/62824264/154159649-7f81c353-41de-4a04-863a-9a9ed9224c3f.JPEG)
        - Your order may be different if you did not revert changes from the visitor account
    2. Go to 'Cards' page in the 'Profile' page
        - Remember the card order and states
        - Change card(s) order and states
        - Go back to the home page and see if those changes are reflected
    3. On the home page, hide some card(s)
        - Go back to the 'Cards' page and see if those changes are reflected
    4. Change the order and states of the cards back to what they were at the start
3. Testing Staff
    1. Upon logging in, go to 'Cards' page in the 'Profile' page and make sure that all the cards are turned on and the order is as follows:
        - ![Screen Shot 2022-02-15 at 2 15 19 PM](https://user-images.githubusercontent.com/62824264/154159693-67322172-232f-43fa-becd-cb07213a4d86.png)
        - Your order may be different if you did not revert changes from the student and/or visitor account
    2. Go to 'Cards' page in the 'Profile' page
        - Change card(s) order and states
        - Go back to the home page and see if those changes are reflected
    3. On the home page, hide some card(s)
        - Go back to the 'Cards' page and see if those changes are reflected
4. Checking if preferences stay
    1. To check if both staff and student preferences stay, make sure you have some changes (order and states) in your student and staff accounts
    2. Delete the app
    3. Download it again
    4. Log in as a student then a staff
        - Check to see if your preferences have saved in both accounts
